torch/distributed/fsdp/_optim_utils.py

```diff
@@ -341,14 +341,14 @@ def _broadcast_processed_state(
     group: Optional[dist.ProcessGroup],
 ) -> Dict[str, Any]:
     objects: List[Any] = [None]
-    if fsdp_state.rank == 0:
+    if dist.get_rank(group) == 0:
```

Comment:
This looks legit since we use `dist.broadcast_object_list(objects, src=0, group=group)` in the next few lines. If fsdp_state.rank mismatches `dist.broadcast_object_list(objects, src=0, group=group)`, the broadcast won't work.